core/kazoo_tasks/src/kz_csv.erl

```diff
@@ -90,32 +95,104 @@ fold(CSV, Fun, Acc)
 %% @doc
 %% @end
 %%--------------------------------------------------------------------
--spec take_row(CSV) -> {row(), CSV} |
-                       'eof' when
-      CSV :: binary().
+-spec take_row(binary()) -> {row(), binary()} |
+                            'eof'.
 take_row(<<>>) -> 'eof';
 take_row(CSV=?NE_BINARY) ->
-    case binary:split(CSV, [<<"\r\n">>, <<"\n\r">>, <<"\r\r">>, <<"\n">>, <<"\r">>]) of
-        [<<>>|_] -> 'eof';
-        [Row] ->
-            {split_row(Row), <<>>};
-        [Row, CSVRest] ->
-            {split_row(Row), CSVRest}
+    case split_row(CSV) of
+        {[], <<>>} -> 'eof';
+        {Row, Rest} -> {Row, Rest}
     end.
 
 %%--------------------------------------------------------------------
 %% @public
 %% @doc
 %% @end
 %%--------------------------------------------------------------------
--spec split_row(ne_binary()) -> row().
+-spec split_row(ne_binary()) -> {row(), binary()}.
 split_row(Row=?NE_BINARY) ->
-    [case Cell of
-         <<>> -> ?ZILCH;
-         _ -> Cell
-     end
-     || Cell <- binary:split(Row, <<$,>>, ['global'])
-    ].
+    split_fields(Row, []).
+
+-spec split_fields(binary(), [binary()]) -> {row(), binary()}.
+split_fields(<<>>, Fields) ->
+    {lists:reverse(Fields), <<>>};
+
+split_fields(<<"\n\r", Rest/binary>>, Fields) ->
+    {lists:reverse(Fields), Rest};
+split_fields(<<"\r\n", Rest/binary>>, Fields) ->
+    {lists:reverse(Fields), Rest};
+split_fields(<<"\r\r", Rest/binary>>, Fields) ->
+    {lists:reverse(Fields), Rest};
+
+split_fields(<<"\n", Rest/binary>>, Fields) ->
+    {lists:reverse(Fields), Rest};
+split_fields(<<"\r", Rest/binary>>, Fields) ->
+    {lists:reverse(Fields), Rest};
+
+split_fields(<<$", Row/binary>>, Fields) ->
+    split_field(Row, $", Fields);
+split_fields(<<$', Row/binary>>, Fields) ->
+    split_field(Row, $', Fields);
+split_fields(Row, Fields) ->
+    split_field(Row, $,, Fields).
+
+-type field_terminator() :: 34 | 39 | 44. %% $" | $' | $,
+
+-spec split_field(binary(), field_terminator(), row()) ->
+                         {row(), binary()}.
+-spec split_field(binary(), field_terminator(), row(), [byte()]) ->
+                         {row(), binary()}.
+split_field(Row, EndChar, Fields) ->
+    split_field(Row, EndChar, Fields, []).
+
+split_field(<<$,>>, $,, Fields, FieldSoFar) ->
+    split_fields(<<>>, [<<>>, iolist_to_binary(lists:reverse(FieldSoFar)) | Fields]);
```

Comment:
But in kz_csv a cell can either be ?ZILCH or ne_binary(). For cells declared as `""` I think the expected value is `<<"\'\'">>` even. No?